bioptim/gui/graph.py

```diff
@@ -0,0 +1,700 @@
+
+        # Draw OCP node
+        G.node("OCP", shape="Mdiamond")
+
+        # Draw nlp clusters and edges
+        for phase_idx in range(self.ocp.n_phases):
+            self._draw_nlp_cluster(G, phase_idx)
+            self._draw_edges(G, phase_idx)
+
+        # Draw phase_transitions
+        self._draw_phase_transitions(G)
+
+        # Draw phaseless objectives
+        self._draw_phaseless_cluster(G)
+        self._draw_phase_trans_to_phaseless_edge(G, self._global_objectives_to_str(self.ocp.J)[0], self.ocp.n_phases)
+
+        # Display graph
+        G.view()
+
+    def _vector_layout(self, vector: list, size: int):
+        """
+        Resize vector content for display task
+
+        Parameters
+        ----------
+        vector: list
+            The vector to be condensed
+        size: int
+            The size of the vector
+        """
+
+        if size > 1:
+            condensed_vector = "[ "
+            count = 0
+            for var in vector:
+                count += 1
+                condensed_vector += f"{float(var)} "
+                if count == 5:
+                    condensed_vector += f"... <br/>... "
+                    count = 0
+            condensed_vector += "]"
+        else:
+            condensed_vector = f"{float(vector[0])}"
+        return condensed_vector
+
+    def _constraint_to_str(self, constraint: Constraint):
+        """
+        Convert constraint information into an easy-to-read string
+
+        Parameters
+        ----------
+        constraint: Constraint
+            The constraint to be converted
+        """
+
+        constraint_str = ""
+        constraint_str += f"{constraint.name}<br/>"
+        constraint_str += f"Min bound: {constraint.min_bound}<br/>"
+        constraint_str += f"Max bound: {constraint.max_bound}<br/>"
+        constraint_str += (
+            f"{f'Target: {constraint.sliced_target} <br/><br/>'}" if constraint.sliced_target is not None else ""
+        )
+        constraint_str += self._add_dict_to_str(constraint.params)
+        return constraint_str
+
+    def _global_objectives_to_str(self, objective_list: ObjectiveList):
+        """
+        Convert global objectives of the ocp to string
+
+        Parameters
+        ----------
+        objective_list: ObjectiveList
+            The list of global objectives to be converted
+        """
+
+        global_objectives = ""
+        global_objectives_names = []
+        for list_objective in objective_list:
+            if len(list_objective) > 0 and isinstance(
+                list_objective[0]["objective"].type, ObjectiveFcn.Mayer or ObjectiveFcn.Lagrange
+            ):
+                objective = list_objective[0]["objective"]
+                global_objectives += f"<b>Objective:</b> {objective.name} <br/>"
+                global_objectives += f"<b>Type:</b> {objective.type} <br/>"
+                global_objectives_names += objective.name
+                global_objectives += (
```

Comment:
I'll stop writing "same" from here, but make sure only very hard to test or obvious lines are not tested. Actual features should be covered

Reply:
Done